.github/workflows/build-image.yaml

```diff
@@ -0,0 +1,48 @@
+name: "Build Docker Image"
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}/fedora-36
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          tags: |
+            type=sha,format=short
+          flavor: |
+            latest=true
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
```

Comment:
I presume we intend to support multiple containers, correct? We may want to go ahead and create some directory structure and use the 'file' input for the specific dockerfile targeted.

Reply:
In that case I would pull over the structure I had in the [other repo](https://github.com/osteffenrh/edk2-build-images) already.

Reply:
I agree, I like the structure of that repo.

Reply:
I think we can change to this when we add more dockerfiles in the future